src/cfnlint/rules/resources/updatepolicy/Configuration.py

```diff
@@ -0,0 +1,221 @@
+                },
+                'MinSuccessfulInstancesPercent': {
+                    'PrimitiveType': 'Integer'
+                },
+                'PauseTime': {
+                    'PrimitiveType': 'String'
+                },
+                'SuspendProcesses': {
+                    'PrimitiveType': 'List',
+                    'ValidValues': [
+                        'Launch', 'Terminate', 'HealthCheck',
+                        'ReplaceUnhealthy', 'AZRebalance', 'AlarmNotification',
+                        'ScheduledActions', 'AddToLoadBalancer'
+                    ]
+                },
+                'WaitOnResourceSignals': {
+                    'PrimitiveType': 'Boolean'
+                }
+            },
+            'AutoScalingScheduledAction': {
+                'IgnoreUnmodifiedGroupSizeProperties': {
+                    'PrimitiveType': 'Boolean'
+                }
+            },
+            'CodeDeployLambdaAliasUpdate': {
+                'AfterAllowTrafficHook': {
+                    'PrimitiveType': 'String'
+                },
+                'ApplicationName': {
+                    'PrimitiveType': 'String'
+                },
+                'BeforeAllowTrafficHook': {
+                    'PrimitiveType': 'String'
+                },
+                'DeploymentGroupName': {
+                    'PrimitiveType': 'String'
+                },
+            }
+        },
+        'main': {
+            'AutoScalingReplacingUpdate': {
+                'Type': 'AutoScalingReplacingUpdate',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'AutoScalingRollingUpdate': {
+                'Type': 'AutoScalingRollingUpdate',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'AutoScalingScheduledAction': {
+                'Type': 'AutoScalingScheduledAction',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'CodeDeployLambdaAliasUpdate': {
+                'Type': 'CodeDeployLambdaAliasUpdate',
+                'ResourceTypes': [
+                    'AWS::Lambda::Alias'
+                ]
+            },
+            'UseOnlineResharding': {
+                'PrimitiveType': 'Boolean',
+                'ResourceTypes': [
+                    'AWS::ElastiCache::ReplicationGroup'
+                ]
+            }
+        }
+    }
+
+    def check_primitive_type(self, value, prim_type, path, **kwargs):
+        """
+            Check primitive types
+        """
+        matches = []
+
+        if isinstance(value, dict):
+            # this could be a ref, getatt, findinmap
+            return matches
+
+        default_message = 'Value for {0} must be of type {1}'
+        if prim_type == 'Boolean':
```

Comment:
you can just put `bool`, `six.string_types` etc. in the dict above, then avoid this longwinded `if` block with

```python
if not isinstance(value, prim_type):
    matches.append(RuleMatch(path, default_message.format(path[-1], prim_type)))
```

sure you get `int` in the message instead of `Integer`, etc., but I think most people would understand what that means

Reply:
I was able to take this advice and clean it up a little bit (I think).  The hard part I ran into is that six.string_types is a tuple and not a class so that was a little harder to print.